### PR TITLE
Add option to run A/B-test on two arbitrary test-report.json files

### PR DESCRIPTION
**Pull request by roypat**
This allows comparisons of two test runs that were not orchestrated by the A/B-testing framework, for example if the two runs were done on different instances. Use this if the data you want to compare is from anything other than "two firecracker revisions tested on the same machine inside the same docker image".

## Changes

...

## Reason

...

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tools/ab_test.py [hunk 1]
```diff
@@ -78,16 +78,12 @@ def extract_dimensions(emf):
     return {key: emf[key] for key in emf if key in dimension_list}
 
 
-def reemit_emf_and_get_data(log_entry: str, revision: str):
+def process_log_entry(emf: dict):
     """Parses the given EMF log entry, and reemits it, overwriting the attached "git_commit_id" field
     with the given revision
```
Comment (pb8o) [inline tools/ab_test.py:83]: I guess the comment does not match the code now?

File: tools/ab_test.py [hunk 2]
```diff
@@ -109,22 +105,30 @@ def find_unit(emf: dict, metric: str):
...
 
-    Also reemits all EMF logs."""
+    If `reemit` is True, it also reemits all EMF logs to a local EMF agent."""
     # Dictionary mapping EMF dimensions to A/B-testable metrics/properties
     processed_emf = {}
 
-    report = json.loads(Path("test_results/test-report.json").read_text("UTF-8"))
+    report = json.loads(report_path.read_text("UTF-8"))
     for test in report["tests"]:
         for line in test["teardown"]["stdout"].splitlines():
```
Comment (pb8o) [inline tools/ab_test.py:119]: Do we have any tests that emit metrics during "call" phase? Could we collect both phases?

File: tools/ab_test.py (hunk 1 above)
Comment (roypat) [reply tools/ab_test.py]: ahh, true, fixed!

File: tools/ab_test.py (hunk 2 above)
Comment (roypat) [reply tools/ab_test.py:119]: We do not currently, because all metrics are emitted by the `metrics` fixture, which only flushes during teardown.

Comment (roypat) [reply tools/ab_test.py:119]: I've had a quick look at whether it'd be an easy change to consider both "teardown" and "call", but it seems that either the "call" key or the "stdout" subkey are not always present :(

**Review by pb8o — APPROVED** [review]

**Review by sudanl0 — APPROVED** [review]

**Pull request merged**